vendor/k8s.io/klog/v2/internal/buffer/buffer.go

```diff
@@ -40,44 +40,22 @@ type Buffer struct {
 	next *Buffer
 }
 
-// Buffers manages the reuse of individual buffer instances. It is thread-safe.
-type Buffers struct {
-	// mu protects the free list. It is separate from the main mutex
-	// so buffers can be grabbed and printed to without holding the main lock,
-	// for better parallelization.
-	mu sync.Mutex
-
-	// freeList is a list of byte buffers, maintained under mu.
-	freeList *Buffer
+var buffers = sync.Pool{
+	New: func() interface{} {
+		return new(Buffer)
+	},
 }
 
 // GetBuffer returns a new, ready-to-use buffer.
-func (bl *Buffers) GetBuffer() *Buffer {
-	bl.mu.Lock()
-	b := bl.freeList
-	if b != nil {
-		bl.freeList = b.next
-	}
-	bl.mu.Unlock()
-	if b == nil {
-		b = new(Buffer)
-	} else {
-		b.next = nil
-		b.Reset()
-	}
+func GetBuffer() *Buffer {
+	b := buffers.Get().(*Buffer)
+	b.Reset()
 	return b
 }
 
 // PutBuffer returns a buffer to the free list.
-func (bl *Buffers) PutBuffer(b *Buffer) {
-	if b.Len() >= 256 {
```

Comment:
pruning big buffers got dropped... is it problematic if the pool fills with large under-utilized buffers? this reminds me of https://github.com/kubernetes/kubernetes/issues/114276 / https://github.com/golang/go/issues/23199

Reply:
I was thinking about this and remained undecided at that time. In the worst case, each buffer instance is filled with one formatted log message. Those shouldn't be that big.

On the other hand, all it takes is one badly written log call that gets hit often enough and sooner or later all buffers will be large enough for that one log call.

Shall I bring back the limit?

Reply:
without data, it's hard to know the impact of dropping this... this is simple enough to preserve that I would probably keep it

Reply:
(and I've seen some egregiously long log lines, so I don't love the idea of retaining buffers sized to fit them in perpetuity)

Reply:
You are right, as usual - I'll bring it back: https://github.com/kubernetes/klog/issues/364

My preference would be to go ahead with this PR in the meantime, to give the update some soak time in the Kubernetes CI (just in  case...).

Reply:
is there a reason not to restore it and do a tag bump now? I'm not sure what the goal of soaking in CI before picking that up is

Reply:
If there are other unexpected issues, I could fix them in the same release - that's all.

The next klog release will be in a few days, I have other things pending for it that need to be reviewed.

Reply:
@liggitt: the next klog release got delayed because there was some other change where the review took a bit longer. Now v2.90.1 is out and this PR is using it - please take another look.